packages/client/src/workflow-client.ts

```diff
@@ -268,6 +299,31 @@ export interface AsyncWorkflowListIterable extends AsyncIterable<WorkflowExecuti
   intoHistories: (intoHistoriesOptions?: IntoHistoriesOptions) => AsyncIterable<HistoryAndWorkflowId>;
 }
 
+/**
+ * A client-side handle to an Update.
+ */
+export interface WorkflowUpdateHandle<Ret> {
+  /**
+   * The ID of this Update request.
+   */
+  id: string;
+
+  /**
+   * The ID of the Workflow being targeted by this Update request.
+   */
+  workflowId: string;
+
+  /**
+   * The ID of the Run of the Workflow being targeted by this Update request.
+   */
+  runId?: string;
```

Comment:
> Also document why this is optional.

Good catch. Fixed to take `runId` from the response (with non-nullish type assertions)

> Could consider also exposing `firstExecutionRunId`

OK, I haven't quite convinced myself that it belongs on the update handle. AIUI the semantics would be "this is the `firstExecutionRunId` you supplied when submitting the update, in case it's useful again for other requests. If it is non-undefined then you know it matches the execution chain of this workflow (because otherwise it would have caused an error in the update request you just did)"